### DISCUSSION TITLE
PNG with transparent background to SVG with transparent background

### DISCUSSION THREAD

**Discussion by codingdudecom** [visioncortex/vtracer#75] [Q&A]
Hi,

I'd like to know if it's possible to transform a PNG with transparent background into an SVG with transparent background. I seem to be getting white/gray color in place of the alpha areas.

**Comment by tyt2y3** [comment]
It is! But it is only available on the CLI. I haven't updated the web interface yet sadly.

**Reply by codingdudecom** [reply]
oh, and how would I do that in the CLI?

**Reply by tyt2y3** [reply]
I think it's enabled by default

**Reply by CptJackieSparrow** [reply]
Unfortunately it's not. Couldn't find it on the readme also. Is there a way?

**Reply by tyt2y3** [reply]
I already updated the web app

**Reply by CptJackieSparrow** [reply]
> I already updated the web app

I was talking about the local one. Thank you for your work mate!

**Comment by codingdudecom** [comment]
so what would it take to get it working in the webapp as well? can you point me in the right direction?

**Reply by tyt2y3** [reply]
The logic for transparent handling hasn't been ported yet.

**Comment by codingdudecom** [comment]
I understand, sorry for bothering you with this, but is this the change that makes transparent background?

https://github.com/visioncortex/vtracer/pull/23/commits/16dfb76835c6e082a6068d3e9144a756761ef304

Is my understanding correct that you are basically 
- finding an unused color
- setting it for the transparent areas
- discarding this color when producing the SVG?

**Reply by tyt2y3** [reply]
Yes! The code has been refactored a bit since then. If you can make a PR, I can review it.

**Comment by tyt2y3** [comment]
I updated the webapp to handle transparent background now
